libcudacxx/include/cuda/__memory_resource/resource_ref.h

```diff
@@ -28,14 +28,49 @@
 #  include <cuda/std/__concepts/_One_of.h>
 #  include <cuda/std/__concepts/all_of.h>
 #  include <cuda/std/__memory/addressof.h>
+#  include <cuda/std/__type_traits/integral_constant.h>
```

Comment:
Nitpick: Is there a place where we use a concrete type from `integral_constant.h` like `true_type`. Otherwise I would like to drop it here. 

If something breaks, we need to include it where needed not in the downstream header